packages/components/src/local/organisms/setting-forces/index.tsx

```diff
@@ -119,17 +119,25 @@ export const SettingForces: React.FC<PropTypes> = ({
         html += `<li>${item}</li>`
         return html
       }, '')
+
       attributeErrors.length > 0 && toggleModal(`The attributes for some assets did not match with type details. These fixes have been applied: <br/> ${attrsbuteErrorList}`)
 
       if (onSave) {
-        forcesData.some(force => {
-          if (force.uniqid === selectedForce.uniqid) {
-            // TODO: should loop each asset and assign only the `attributeValues` for each one instead of assign the whole asset to force?
-            force.assets = selectedForce.assets
-            return true
-          }
-          return false
-        })
+        // if the data is wrong and has been modified, should update back to the forceData
+        // If not, just save the forcesData
+        if (attributeErrors.length) {
+          forcesData.some(force => {
+            if (force.uniqid === selectedForce.uniqid && force.assets) {
+              force.assets.forEach((asset, idx) => {
+                if (selectedForce.assets) {
+                  asset.attributeValues = selectedForce.assets[idx].attributeValues
+                }
+              })
+              return true
```

Comment:
if we are using `some` won't this code drop out on the first `true` value?  My guess is that we should run through all of the forces.